source/language/openpulse.rst

```diff
@@ -290,28 +298,262 @@ pushing the results into some buffer which is then accessed outside the program.
 Timing
 ------
 
-Each frame maintains its own "clock". When a pulse is played the clock for 
-that frame advances by the length of the pulse. 
+Each frame maintains its own "clock". When a pulse is played the clock for
+that frame advances by the length of the pulse.
 
-For frames, everything behaves analogous to qubits in the 
+For frames, everything behaves analogous to qubits in the
 `Delays <delays.html>`_ section of this specification. There are however some
 small differences.
 
-The ``delay`` instruction may take a frame instead of a qubit. The ``barrier`` 
-instruction may also take a list of frames intead of a list of qubits.
+The ``delay`` instruction may take a frame instead of a qubit. The ``barrier``
+instruction may also take a list of frames intead of a list of qubits and aligns the time
+of the clocks given as arguments.
+
+``defcal`` blocks have an implicit ``barrier`` on every frame that enters the block,
+meaning that those clocks are guaranteed to be aligned at the start of the block.
+These blocks also need to have a well-defined length, similar to the ``boxas`` block.
 
-``defcal`` blocks have an implicit barrier on every frame used within the block,
-meaning that clocks are guaranteed to be aligned at the start of the block.
-These blocks also need to have a well-defined length, similar to the ``boxas``
-block.
+.. code-block:: javascript
 
-.. code-block:: none
+   waveform p = ...; // some 100dt waveform
 
-   pulse p = ...; // some 100dt pulse
+   defcal prelude {
+     frame driveframe1 = Frame(1.0, 5e9, 0.0);
+     frame driveframe2 = Frame(1.0, 6e9, 0.0);
+   }
 
-   defcal simultaneous_pulsed_gate %0 {
-     play(p) frameof("drive0", %0);
-     delay[20dt] frameof("drive1", %0);
-     // Starts the 100dt pulse 20dt into "drive0" already playing it
-     play(p) frameof("drive1");
+   defcal aligned_gates {
+     // driveframe1 and driveframe2 used in this defcal, so clocks are aligned
+     play(tx0, p, driveframe1);
+     delay[20dt] driveframe;
+     // Clocks now unaligned by 120dt, so we use a `barrier` to re-align
+     barrier(driveframe1, driveframe2);
+     // `driveframe2` will now play a pulse 120dt after `driveframe1` finishes playing
+     play(tx0, p, driveframe2);
    }
+
+Examples
+--------
+
+Cross-resonance gate
+~~~~~~~~~~~~~~~~~~~~
+
+
+.. code-block:: javascript
+
+  defcal prelude {
+     frame frame0 = Frame(1.0, 5e9, 0.0);
+  }
+
+  defcal cross_resonance %0 %1 {
```

Comment:
We've switched to using $q for physical qubits, as `%` refers to the modulo operator. See https://qiskit.github.io/openqasm/language/pulses.html